libsolidity/ast/ASTJsonConverter.cpp

```diff
@@ -252,14 +255,20 @@ bool ASTJsonConverter::visit(ImportDirective const& _node)
 
 bool ASTJsonConverter::visit(ContractDefinition const& _node)
 {
+	std::vector<ContractDefinition const*> contractDeps{
+		_node.annotation().contractDependencies.begin(),
+		_node.annotation().contractDependencies.end()
+	};
+	std::sort(contractDeps.begin(), contractDeps.end());
```

Comment:
This still does not fix it, they dependencies will be sorted by address, which is not consistent.

Maybe add a parameter to `getContainerIds` which allows you to sort by ID value?